test/e2e/performanceprofile/functests/2_performance_update/memorymanager.go

```diff
@@ -0,0 +1,851 @@
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// MMPod Memory Manager Pod Definition
+type MMPod struct {
+	podV1Struct                   *corev1.Pod
+	namespace                     string
+	cpu, memory, noOfhpgs, medium string
+	hpgSize                       performancev2.HugePageSize
+	nodeSelector                  map[string]string
+	runtimeclass                  string
+}
+
+var _ = Describe("[rfe_id: 43186][memorymanager] Memorymanager feature", func() {
+	var workerRTNodes []corev1.Node
+	var profile, initialProfile *performancev2.PerformanceProfile
+	var performanceMCP string
+	var err error
+
+	Context("Group Both Numa Nodes with restricted topology", Ordered, func() {
+		var numaCoreSiblings map[int]map[int][]int
+		var reserved, isolated []string
+		var hpCount = 20
+		testutils.CustomBeforeAll(func() {
+			var policy = "restricted"
+			workerRTNodes = getUpdatedNodes()
+			profile, err = profiles.GetByNodeLabels(testutils.NodeSelectorLabels)
+			Expect(err).ToNot(HaveOccurred())
+			performanceMCP, err = mcps.GetByProfile(profile)
+			Expect(err).ToNot(HaveOccurred())
+
+			// Save the original performance profile
+			initialProfile = profile.DeepCopy()
+
+			for _, node := range workerRTNodes {
+				numaInfo, err := nodes.GetNumaNodes(&node)
+				Expect(err).ToNot(HaveOccurred())
+				if len(numaInfo) < 2 {
+					Skip(fmt.Sprintf("This test need 2 Numa nodes. The number of numa nodes on node %s < 2", node.Name))
+				}
+			}
+
+			By("Modifying Profile")
+			for _, node := range workerRTNodes {
+				numaCoreSiblings, err = nodes.GetCoreSiblings(&node)
+			}
+			// Get cpu siblings from core 0, 1
+			for reservedCores := 0; reservedCores < 2; reservedCores++ {
+				cpusiblings := nodes.GetAndRemoveCpuSiblingsFromMap(numaCoreSiblings, reservedCores)
+				reserved = append(reserved, cpusiblings...)
+			}
+			reservedCpus := strings.Join(reserved, ",")
+			for key := range numaCoreSiblings {
+				for k := range numaCoreSiblings[key] {
+					cpusiblings := nodes.GetAndRemoveCpuSiblingsFromMap(numaCoreSiblings, k)
+					isolated = append(isolated, cpusiblings...)
+				}
+			}
+			isolatedCpus := strings.Join(isolated, ",")
+			reservedSet := performancev2.CPUSet(reservedCpus)
+			isolatedSet := performancev2.CPUSet(isolatedCpus)
+
+			hpSize1G := performancev2.HugePageSize("1G")
+			hpSize2M := performancev2.HugePageSize("2M")
+
+			requiredHugepages := &performancev2.HugePages{
+				DefaultHugePagesSize: &hpSize1G,
+				Pages: []performancev2.HugePage{
+					{
+						Count: int32(hpCount),
+						Size:  hpSize2M,
+					},
+				},
+			}
+			profile.Spec.HugePages = requiredHugepages
+
+			if !*profile.Spec.RealTimeKernel.Enabled {
+				profile.Spec.RealTimeKernel = &performancev2.RealTimeKernel{
+					Enabled: pointer.BoolPtr(true),
+				}
+			}
+			if profile.Spec.WorkloadHints == nil {
```

Comment:
This might cause subtle bug in case `profile.Spec.WorkloadHints != nil` but  `RealTime` is `false`, it's probably better to override it all the time.